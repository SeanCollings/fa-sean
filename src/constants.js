export const API_THE_ICONIC_URL =
  'https://eve.theiconic.com.au/catalog/products';

export const FILTER_SORT = {
  popularity: { display: 'Popularity', field: 'popularity' },
  new: { display: 'New Arrivals', field: 'new' },
  price_high: { display: 'Price High to Low', field: 'price_high' },
  price_low: { display: 'Price Low to High', field: 'price_low' },
  brand_asc: { display: 'Brand A to Z', field: 'brand_asc' },
  brand_desc: { display: 'Brand Z to A', field: 'brand_desc' },
};

export const FILTER_GENDER = {
  female: { display: 'Women', field: 'female' },
  male: { display: 'Men', field: 'male' },
  kids: { display: 'Kids', field: 'kids' },
};

export const COLOURS = {
  dark: '#232323',
  gray: '#eee',
  light: '#f7f7f7;',
  textDark: '#4d4d4d',
  textGray: '#909090',
  red: '#ef423c',
  white: '#ffffff',
  errorText: '#ee110d',
  errorBackground: '#fcecec',
};

// Â»
