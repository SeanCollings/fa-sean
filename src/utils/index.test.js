import { getFilterProperties, getProductUrl } from '.';
import { FILTER_GENDER, FILTER_SORT } from '../constants';

describe('utils', () => {
  describe('getProductUrl', () => {
    test('should return product url with default paramaters', () => {
      const url = getProductUrl({});
      expect(url).toMatchSnapshot();
    });

    test('should return product url with updated gender paramater', () => {
      const url = getProductUrl({ gender: FILTER_GENDER.male.field });
      expect(url).toMatchSnapshot();
    });

    test('should return product url with updated sort paramater', () => {
      const url = getProductUrl({ sort: FILTER_SORT.price_high.field });
      expect(url).toMatchSnapshot();
    });
  });

  describe('getFilterProperties', () => {
    const state = {
      filter: {
        gender: FILTER_GENDER.male,
        sort: FILTER_SORT.price_low,
      },
    };

    test('should return an empty object if no state is passed in', () => {
      const properties = getFilterProperties({});
      expect(properties).toMatchSnapshot();
    });

    test('should return initial state if no properties passed in', () => {
      const properties = getFilterProperties({ state });
      expect(properties).toMatchSnapshot();
    });

    test('should return updated gender property', () => {
      const properties = getFilterProperties({
        gender: FILTER_GENDER.kids,
        state,
      });
      expect(properties).toMatchSnapshot();
    });

    test('should return updated sort property', () => {
      const properties = getFilterProperties({
        sort: FILTER_SORT.brand_asc,
        state,
      });
      expect(properties).toMatchSnapshot();
    });
  });
});
