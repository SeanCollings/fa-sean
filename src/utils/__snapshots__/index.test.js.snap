// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils getFilterProperties should return an empty object if no state is passed in 1`] = `Object {}`;

exports[`utils getFilterProperties should return initial state if no properties passed in 1`] = `
Object {
  "gender": Object {
    "display": "Men",
    "field": "male",
  },
  "sort": Object {
    "display": "Price Low to High",
    "field": "price_low",
  },
}
`;

exports[`utils getFilterProperties should return updated gender property 1`] = `
Object {
  "gender": Object {
    "display": "Kids",
    "field": "kids",
  },
  "sort": Object {
    "display": "Price Low to High",
    "field": "price_low",
  },
}
`;

exports[`utils getFilterProperties should return updated sort property 1`] = `
Object {
  "gender": Object {
    "display": "Men",
    "field": "male",
  },
  "sort": Object {
    "display": "Brand A to Z",
    "field": "brand_asc",
  },
}
`;

exports[`utils getProductUrl should return product url with default paramaters 1`] = `"https://eve.theiconic.com.au/catalog/products?gender=female&page=1&page_size=10&sort=popularity"`;

exports[`utils getProductUrl should return product url with updated gender paramater 1`] = `"https://eve.theiconic.com.au/catalog/products?gender=male&page=1&page_size=10&sort=popularity"`;

exports[`utils getProductUrl should return product url with updated sort paramater 1`] = `"https://eve.theiconic.com.au/catalog/products?gender=female&page=1&page_size=10&sort=price_high"`;
